### PR TITLE
[Conversion gRPC-REST] Classifier endpoint

### PR DESCRIPTION
**Pull request by pauloabreu**
# Resume

This Pull Request create a new endpoint, called Classifer.

- Can return the classifiers from a org
- Can delete a classifier
- And have a generic crud

### CODE REVIEW

**Comment by Sandro-Meireles** [comment]
This package needs to be inside the [internal](https://github.com/Ilhasoft/rapidpro-apps/tree/main/weni/internal) app.
See that there are 2 apps: `flows` and `orgs`. make your package based on them.

**Comment by Sandro-Meireles** [comment]
Use the abstractions that the rest-framework itself has, recreating pre-established behaviors in addition to being redundant increases the chance of errors. `ClassifierEndpoint` class has redundant methods, refactor what is possible.

**Comment by Sandro-Meireles** [comment]
Add a line at the end of each file

**Comment by Sandro-Meireles** [comment]
Remove imports that are not being used

**Comment by Sandro-Meireles** [comment]
the code is duplicated in flows/classifiers and internal/classifiers?

**Comment by Sandro-Meireles** [comment]
Replace everything `JsonResponse` with `Response`

File: weni/flows/classifier/serializers.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    class Meta:
+        model = Classifier
+        proto_class = classifier_pb2.Classifier
+        fields = ["uuid", "is_active", "classifier_type", "name", "access_token", "org", "user"]
+
+
+class ClassifierDeleteSerializer(serializers.Serializer):
+    uuid = serializers.UUIDField(required=True)
+    user = weni_serializers.UserEmailRelatedField(required=True)
```
Comment (elitonzky) [inline weni/flows/classifier/serializers.py:35]: there is a last blank line missing at the end of the code

File: weni/flows/classifier/tests.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+from temba.tests import TembaTest, mock_mailroom
+from temba.orgs.models import Org
+from temba.classifiers.models import Classifier, Intent
+from temba.classifiers.types.wit import WitType
+from temba.classifiers.types.luis import LuisType
+from weni.protobuf.flows import classifier_pb2, classifier_pb2_grpc
+
+
+
```
Comment (elitonzky) [inline weni/flows/classifier/tests.py:21]: remove a blank line, keep only two before class

File: weni/flows/classifier/tests.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        )
+
+        self.org = Org.objects.create(name="Weni", timezone="America/Maceio", created_by=self.admin, modified_by=self.admin)
+
+        super().setUp()
+
+    def test_destroy_classifier_by_valid_uuid(self):
+        classifier = Classifier.create(self.org, self.admin, LuisType.slug, "Test2", self.config, sync=False)
+        Intent.objects.create(classifier=classifier, name="Test Intent", external_id="FakeExternal")
+
```
Comment (elitonzky) [inline weni/flows/classifier/tests.py:215]: trailing whitespace  at line 215

File: weni/flows/classifier/tests.py [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+class TembaRequestMixin(ABC):
+    def reverse(self, viewname, kwargs=None, query_params=None):
+        url = reverse(viewname, kwargs=kwargs)
+
+        if query_params:
+            return "%s?%s" % (url, urlencode(query_params))
+        else:
+            return url
+
```
Comment (elitonzky) [inline weni/flows/classifier/tests.py:30]: There would be no need for "else" in this case, you could write it as follows.

if query:
____ return query

return url

File: weni/internal/classifier/tests.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+from temba.tests import TembaTest, mock_mailroom
+from temba.orgs.models import Org
+from temba.classifiers.models import Classifier, Intent
+from temba.classifiers.types.wit import WitType
+from temba.classifiers.types.luis import LuisType
+from weni.protobuf.flows import classifier_pb2, classifier_pb2_grpc
+
+
+
```
Comment (elitonzky) [inline weni/internal/classifier/tests.py:21]: Before class, just two blank lines, in this case remove one.

File: weni/internal/classifier/tests.py [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+class ClassifierTestCase(TembaTest, TembaRequestMixin):
+
+    def setUp(self):
+        self.config = {"access_token": "hbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9"}
+
+        self.admin = User.objects.create_user(
+            username="testuser", password="123", email="test@weni.ai", is_superuser=True
+        )
+
+        print(self.admin.is_authenticated)
```
Comment (elitonzky) [inline weni/internal/classifier/tests.py:70]: Remove print line

File: weni/internal/classifier/tests.py [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+
+        self.org = Org.objects.create(name="Weni", timezone="America/Maceio", created_by=self.admin, modified_by=self.admin)
+
+        super().setUp()
+
+    def test_list_classifier(self):
+        org = Org.objects.first()
+        org_uuid = str(org.uuid)
+
+
```
Comment (elitonzky) [inline weni/internal/classifier/tests.py:80]: two blank lines

File: weni/internal/classifier/tests.py [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+        )
+
+        self.org = Org.objects.create(name="Weni", timezone="America/Maceio", created_by=self.admin, modified_by=self.admin)
+
+        super().setUp()
+
+    def test_destroy_classifier_by_valid_uuid(self):
+        classifier = Classifier.create(self.org, self.admin, LuisType.slug, "Test2", self.config, sync=False)
+        Intent.objects.create(classifier=classifier, name="Test Intent", external_id="FakeExternal")
+
```
Comment (elitonzky) [inline weni/internal/classifier/tests.py:219]: trailing whitespace at line 219

File: weni/flows/classifier/tests.py [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+        self.assertEqual(len(response), 1)
+
+        response = response[0]
+
+        self.assertEqual(response.get("name"), "Test")
+        self.assertEqual(response.get("classifier_type"), LuisType.slug)
+        self.assertEqual(response.get("uuid"), str(classifier.uuid))
+        self.assertEqual(response.get("access_token"), self.config["access_token"])
+
+
```
Comment (elitonzky) [inline weni/flows/classifier/tests.py:116]: line break excess

File: weni/flows/classifier/views.py [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        }
+
+        serializer = ClassifierDeleteSerializer(data=data)
+        if not serializer.is_valid():
+            return JsonResponse(data=serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        classifier = Classifier.objects.get(uuid=uuid)
+        classifier.release(serializer.validated_data.get("user"))
+
+        return Response(status=status.HTTP_200_OK)
```
Comment (elitonzky) [inline weni/flows/classifier/views.py:91]: missing line break at end of code

File: weni/flows/classifier/views.py [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        try:
+            classifier = Classifier.objects.get(uuid=uuid)
+        except Classifier.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        return JsonResponse(data=self.get_serializer(classifier).data, status=status.HTTP_200_OK)
+
+    
+    def destroy(self, request, uuid=None):
+
```
Comment (elitonzky) [inline weni/flows/classifier/views.py:78]: trailing-whitespace at line 78

File: weni/flows/classifier/views.py [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+        filters = {}
+
+        if org_uuid is not None:
+            try:
+                org = Org.objects.get(uuid=org_uuid)
+                filters["org"] = org
+            except Org.DoesNotExist:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+
```
Comment (elitonzky) [inline weni/flows/classifier/views.py:44]: trailing-whitespace at line 44

File: weni/flows/classifier/urls.py [hunk 13]
```diff
@@ -0,0 +1,10 @@
+from django.conf.urls import url
+from rest_framework.urlpatterns import format_suffix_patterns
+from rest_framework import routers
+
+from .views import ClassifierEndpoint
+
+router = routers.SimpleRouter()
+router.register("flows-backend/classifier", ClassifierEndpoint, basename="api.v2.flows_backend.classifier")
+
+urlpatterns = format_suffix_patterns(router.urls, allowed=["json", "api"])
```
Comment (elitonzky) [inline weni/flows/classifier/urls.py:10]: missing line break at end of code

File: weni/internal/classifier/tests.py [hunk 14]
```diff
@@ -0,0 +1,229 @@
...
+        self.assertEqual(len(response), 1)
+
+        response = response[0]
+
+        self.assertEqual(response.get("name"), "Test")
+        self.assertEqual(response.get("classifier_type"), LuisType.slug)
+        self.assertEqual(response.get("uuid"), str(classifier.uuid))
+        self.assertEqual(response.get("access_token"), self.config["access_token"])
+
+
```
Comment (elitonzky) [inline weni/internal/classifier/tests.py:120]: The code block above contains excess line breaks:
more details in: 
https://peps.python.org/pep-0008/#blank-lines

**Review by elitonzky — COMMENTED** [review]
small corrections and corrections to pep8

**Pull request closed without merging**